### ISSUE TITLE
Wrap Properties is unavailable in recoverable syntax failure conditions

### ISSUE THREAD

**Issue by 333fred** [dotnet/roslyn#38986]
```cs
    public class Base
    {
        public Base(int i1, int i2)
        {
        }
    }

    public class Derived : Base
    {
        public Derived(int i1, int i2, int i3) : base (,)
        {

        }
    }
```

If I invoke `ctrl+.` on the parameter list of `Derived` here, I do not get offered a wrap parameter quick fix. Same for an incomplete method where the brackets are missing, but the signature is otherwise there. I understand that we're likely just bailing out in the presence of syntax errors, but it would be nice if we could be a bit smarter here, as offering the wrapping, at least in this scenario, is totally possible and, I think, has a well-defined output.

**Comment by CyrusNajmabadi** [comment]
Yes.  Currently we bail out if our parent has any syntax errors.   The presumption is that we won't be able to tell how to wrap properly in that case.

What we likely should do is only bail out if our parent has syntax errors between the beginning of it through the end of hte list that we're wrapping.  i.e. here:

```c#
        [|public Derived(int i1, int i2, int i3)|] : base (,)
        {

        }
```

Syntax errors *after* the construct to be wrapped are unlikely to ever be an issue and can be ignored.
